src/middlewared/middlewared/plugins/pool.py

```diff
@@ -778,12 +779,33 @@ async def do_create(self, job, data):
         asyncio.ensure_future(self.restart_services())
 
         pool = await self.get_instance(pool_id)
-        await self.middleware.call_hook('pool.post_create_or_update', pool=pool)
-        await self.middleware.call_hook(
-            'dataset.post_create', {'encrypted': bool(encryption_dict), **encrypted_dataset_data}
-        )
+        if not await self.middleware.call('failover.licensed'):
```

Comment:
Can it be better to deal with this in the specific hook itself?

Reply:
I got this error and didn't want to mess with it.
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/middlewared/main.py", line 1074, in _call_hook_base                                                                                                                                                                                                                               yield hook, hook['method'](self, *args, **kwargs)                                                                                                                                                                                                                                                                        TypeError: hook_sync_geli() got an unexpected keyword argument 'ha_sync'
```

Reply:
When trying to do these changes inside the hook itself.